src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder salaryInfo = new StringBuilder();
+        LocalDate startDate = LocalDate.parse(dateFrom);
+        LocalDate endDate = LocalDate.parse(dateTo);
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+        salaryInfo.append("Report for period ");
+        salaryInfo.append(dateFrom);
+        salaryInfo.append(" - ");
+        salaryInfo.append(dateTo);
+        for (String name : names) {
+            int salary = 0;
+            for (String datum : data) {
+                String[] lines = datum.split(" ");
+                LocalDate date = LocalDate.parse(lines[0]);
+                String nameUser = lines[1];
+                if (date.isAfter(startDate) && date.isBefore(endDate)
+                        && name.equals(nameUser)) {
+                    salary += Integer.parseInt(lines[2]) * Integer.parseInt(lines[3]);
+                } else if (date.isEqual(startDate) && name.equals(nameUser)
+                        || date.isEqual(endDate) && name.equals(nameUser))  {
```

Comment:
The logic for checking if a date is within the range can be simplified. Instead of checking if the date is after the start date and before the end date, or if it is equal to either, you can use: `if (!date.isBefore(startDate) && !date.isAfter(endDate) && name.equals(nameUser))`.